### DISCUSSION TITLE
How can I effectively use the selenium base equivalent of element.find_element()?

### DISCUSSION THREAD

**Discussion by nwest-encoura** [seleniumbase/SeleniumBase#1828] [General]
I am refactoring some UI tests which were built with raw selenium. 
The solution uses a page object model which largely returns web elements which can then do things like click or send_keys. 

The problem with this is that I have an element, lets call it `card`,  which I get by using a generic locator with `card = driver.find_elements(By.ID, "card")[0]`.

I can then use this web element to click on the close button for example `card.find_element(By.ID, "close-btn")`. 

Is there a selenium base equivalent that lets me do a similar thing? Right now selenium base's `find_elements` returns web elements that must then utilize seleniums built in `find_element` methods, but I would much rather use selenium bases `find_element` method, and forego having to maintain a mixture of locators and methods.

**Comment by mdmintz** [comment]
Hello. In your example, it appears you're trying to find multiple elements with the same ``id``, but the ``id`` should be unique to a web page, meaning that there shouldn't be more than one element with the same ``id``. Second, instead of finding an element inside an element, just use a selector that matches the inner element that you're trying to find. Once you find a list of elements, each item is a ``WebElement`` object, which uses raw Selenium methods.

Note that SeleniumBase also has some list methods, (eg. ``self.click_visible_elements(SELECTOR)``), which will click on every matching element.

**Reply by nwest-encoura** [reply]
I think I used ID just as an example. Most of the time I have to locate elements by css selector which are not unique to each element I am interacting with. It would be nice if selenium base had a custom class for `WebElements` so that we could do things such as `sb.find_element("locator").click("locator"`), currently I believe you need to do `sb.find_element("locator").find_element(By.CSS_SELECTOR, "locator").click()`

**Reply by mdmintz** [reply]
Once you've found a WebElement, you have to use the WebElement methods. But every element on a page should be able to be found with a unique selector, without having to do two different find actions.
